aws-predictions-tfl/src/main/java/com/amplifyframework/predictions/service/TFLiteTextClassificationService.java

```diff
@@ -0,0 +1,332 @@
+     * dictionary["<START>"] = 1    mark for the start of a sentence
+     * dictionary["<UNKNOWN>"] = 2  mark for unknown words (OOV)
+     */
+    private static final String PAD = "<PAD>";
+    private static final String START = "<START>";
+    private static final String UNKNOWN = "<UNKNOWN>";
+
+    private final Map<String, Integer> dictionary = new HashMap<>();
+    private final List<String> labels = new ArrayList<>();
+
+    private final Context context;
+
+    private Interpreter tflite;
+    private AtomicBoolean loaded;
+
+    /**
+     * Constructs an instance of service to perform text
+     * sentiment interpretation using Tensorflow Lite
+     * interpreter.
+     * @param context the Android context for loading model
+     */
+    TFLiteTextClassificationService(Context context) {
+        this.context = context;
+        this.loaded = new AtomicBoolean(false);
+
+        try {
+            // Try loading assets now if possible
+            loadIfNotLoaded();
+        } catch (PredictionsException exception) {
+            // Ignore if it fails to load during configuration.
+
+            // This may sound weird, but we want the model to be loaded in as soon as possible.
+            // But we don't want to throw an error that needs to be caught at configuration time,
+            // since these models are supposed to be optional.
+        }
+    }
+
+    /**
+     * Gets the associated service key of this service for
+     * identification.
+     * @return the service key
+     */
+    @NonNull
+    String getServiceKey() {
+        return SERVICE_KEY;
+    }
+
+    /**
+     * Classifies text to analyze associated sentiments.
+     * @param text the text to classify
+     * @param onSuccess notified when classification succeeds
+     * @param onError notified when classification fails
+     */
+    void classify(
+            @NonNull String text,
+            @NonNull Consumer<InterpretResult> onSuccess,
+            @NonNull Consumer<PredictionsException> onError
+    ) {
+        final Sentiment sentiment;
+        try {
+            loadIfNotLoaded();
+            sentiment = fetchSentiment(text);
+        } catch (PredictionsException exception) {
+            onError.accept(exception);
+            return;
+        }
+
+        InterpretResult result = InterpretResult.builder()
+                .sentiment(sentiment)
+                .build();
+        onSuccess.accept(result);
+    }
+
+    private Sentiment fetchSentiment(String text) throws PredictionsException {
+        float[][] input;
+        float[][] output;
+
+        // Pre-process input text
+        try {
+            input = tokenizeInputText(text);
+            output = new float[1][labels.size()];
+        } catch (IllegalArgumentException exception) {
+            throw new PredictionsException(
+                    "Tensorflow Lite failed to make inference.",
+                    exception,
+                    "Verify that the assets are loaded."
```

Comment:
Which assets?

Reply:
It's actually the wrong message here! Thanks for catching it :)